dace/transformation/passes/analysis/analysis.py

```diff
@@ -333,6 +333,79 @@ def apply_pass(self, top_sdfg: SDFG, _) -> Dict[int, Dict[str, Set[SDFGState]]]:
         return top_result
 
 
+@properties.make_properties
+@transformation.explicit_cf_compatible
+class FindSingleUseData(ppl.Pass):
+    """
+    For each SDFG find all data descriptors that are referenced in exactly one location.
+
+    In addition to the requirement that there exists exactly one AccessNode that
+    refers to a data descriptor the following conditions have to be meet as well:
+    - The data is not read on an interstate edge.
+    - The data is not accessed in the branch condition, loop condition, etc. of
+        control flow regions.
+    - There must be at least one AccessNode that refers to the data. I.e. if it exists
+        inside `SDFG.arrays` but there is no AccessNode, then it is _not_ included.
+
+    It is also important to note that the degree of the AccessNodes are ignored.
+    """
+
+    CATEGORY: str = 'Analysis'
+
+    def modifies(self) -> ppl.Modifies:
+        return ppl.Modifies.Nothing
+
+    def should_reapply(self, modified: ppl.Modifies) -> bool:
+        # If anything was modified, reapply
+        return modified & ppl.Modifies.AccessNodes & ppl.Modifies.CFG
+
+    def apply_pass(self, sdfg: SDFG, _) -> Dict[SDFG, Set[str]]:
+        """
+        :return: A dictionary mapping SDFGs to a `set` of strings containing the name
+            of the data descriptors that are only used once.
+        """
+        # TODO(pschaad): Should we index on cfg or the SDFG itself.
```

Comment:
For something related to data containers and the question of 'can I remove this' I think it makes sense to index on the SDFG level